frontend/src/pages/QueryBuilderChooser/QueryBuilderChooser.tsx

```diff
@@ -0,0 +1,134 @@
+  useEffect(() => {
+    localForage.getItem<string>(localForageKeys.PREFERENCES).then((key) => {
+      if (key) {
+        setChoice(key || undefined);
+        if (key === 'advanced') {
+          history.push('/queries/build/advanced/');
+        } else {
+          history.push('/queries/build/basic/');
+        }
+      }
+    });
+  }, []);
+
+  const handleChange = () => {
+    setChecked(!checked);
+  };
+
+  const handleSave = () => {
+    localForage.getItem<string>(localForageKeys.API_KEY).then((token) => {
+      const userPreference = {
+        preferences: selectedOption,
+        global_choice: false,
+      };
+      axios
+        .request({
+          method: 'post',
+          url: api.routes.USERPREFERENCE,
+          headers: {
+            'Content-Type': 'application/json',
+            Authorization: `token ${token}`,
+          },
+          data: userPreference,
+        })
+        .then(() => {
+          if (selectedOption || choice) {
+            if (selectedOption === 'basic' || choice === 'basic') {
+              history.push('/queries/build/basic/');
+            } else {
+              history.push('/queries/build/advanced/');
+            }
+          }
+        })
+        .catch((err) => console.log(err));
+    });
+    if (checked === true) {
+      localForage.setItem(localForageKeys.PREFERENCES, selectedOption);
+    }
+  };
+  const alertMessage = () => {
+    if (selectedOption === 'basic') {
+      return (
+        <p style={{ fontSize: 14 }}>
+          The original Query Builder. Creates queries using interconnected steps - no SQL knowledge
+          required.
+        </p>
+      );
+    } else {
+      return (
+        <p style={{ fontSize: 14 }}>
+          An improved UI with a lot more flexibility & options for creating advanced queries. Some
+          of its features require a little SQL knowledge.
+        </p>
+      );
+    }
+  };
+
+  return (
+    <div>
+      <Modal show={showModal} onHide={toggleModal}>
+        <Modal.Header closeButton>
+          <Modal.Title>Select Query Builder</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <IRadio
+            variant="danger"
+            id="basic"
+            name="querybuilder"
+            label="Basic"
+            onChange={onRadioChange}
+            inline
+            checked={selectedOption === 'basic'}
+          />
+          <IRadio
+            variant="danger"
+            id="advanced"
+            name="querybuilder"
+            label="Advanced"
+            onChange={onRadioChange}
+            inline
+            checked={selectedOption === 'advanced'}
+          />
+          <Alert variant="secondary" className="p-2">
+            {alertMessage()}
+          </Alert>
+        </Modal.Body>
+        <Modal.Footer>
+          <CheckBox label="Remember my choice" checked={checked} onChange={handleChange} />
+          <Button onClick={handleSave} variant="btn-danger" className="btn-danger">
```

Comment:
Ah, I wondered why every time I clicked `Go to Query Builder` a new preference was added to the database ... my user now has like 20 preferences. Clicking this button shouldn't automatically save. It should only save if a user has chosen  `Remember my choice`. Therefore, I suggest you create a different function for this button that automatically navigates to the user's choice without saving IF they haven't chosen to remember their choice. Else, save.